drivers/clock_control/clock_control_nrf.c

```diff
@@ -530,6 +534,21 @@ static int clk_init(const struct device *dev)
 		    nrfx_isr, nrfx_power_clock_irq_handler, 0);
 	irq_enable(DT_INST_IRQN(0));
 
+#if CONFIG_SOC_NRF5340_CPUAPP
```

Comment:
`nrf_gpio_pin_mcu_select()` can be called only from secure code, so I think you need here:
```
#if defined(CONFIG_SOC_NRF5340_CPUAPP) && \
	!defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
```

Reply:
@anangl I wonder how this could be more future proof, considering we may have multiple SoC variants.
Something around this, perhaps: 

```
#if defined(CONFIG_SOC_SERIES_NRF53X) && \
    defined(CONFIG_ARM_TRUSTZONE_M) && \
    !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
```

what do you think?

Reply:
This code uses pin locations specific to nRF5340, and the whole thing is considered a temporary solution (see the PR description), so I'd rather not try to make it more generic now.

Reply:
Sure, it was just a thought. But even if this ends up in the nrfx clock driver, we will need to address the challenge of a future-proof solution